modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -0,0 +1,427 @@
+          _id_map.insert(std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
+              pin->first, pin->second));
+        else if (pin->second != _id_map.find(pin->first)->second)
+          mooseError("Multiple region definitions for the same pin type. Check pin_type ids.\n");
+      }
+
+      if (_geom_type == "Hex")
+      {
+        subdomain_id_type assembly_type =
+            getMeshProperty<subdomain_id_type>("assembly_type_id", assembly);
+        if (_background_id_map.find(assembly_type) == _background_id_map.end())
+        {
+          std::vector<subdomain_id_type> background_regions =
+              getMeshProperty<std::vector<subdomain_id_type>>("background_region_ids", assembly);
+          std::vector<std::vector<subdomain_id_type>> duct_regions =
+              getMeshProperty<std::vector<std::vector<subdomain_id_type>>>("duct_region_ids",
+                                                                           assembly);
+          _background_id_map.insert(std::pair<subdomain_id_type, std::vector<subdomain_id_type>>(
+              assembly_type, background_regions));
+          _duct_id_map.insert(
+              std::pair<subdomain_id_type, std::vector<std::vector<subdomain_id_type>>>(
+                  assembly_type, duct_regions));
+        }
+      }
+    }
+  }
+  declareMeshProperty("pin_id_map", _id_map);
+
+  declareMeshProperty("assembly_pitch", getMeshProperty<Real>("assembly_pitch", _reactor_params));
+
+  if (_extrude && _mesh_dimensions == 3)
+  {
+    std::vector<Real> axial_boundaries =
+        getMeshProperty<std::vector<Real>>("axial_boundaries", _reactor_params);
+    const auto top_boundary = getMeshProperty<boundary_id_type>("top_boundary_id", _reactor_params);
+    const auto bottom_boundary =
+        getMeshProperty<boundary_id_type>("bottom_boundary_id", _reactor_params);
+    {
+      declareMeshProperty("extruded", true);
+      auto params = _app.getFactory().getValidParams("FancyExtruderGenerator");
+
+      params.set<MeshGeneratorName>("input") =
+          _empty_pos ? name() + "_deleted" : name() + "_pattern";
+      params.set<Point>("direction") = Point(0, 0, 1);
+      params.set<std::vector<unsigned int>>("num_layers") =
+          getMeshProperty<std::vector<unsigned int>>("axial_mesh_intervals", _reactor_params);
+      params.set<std::vector<Real>>("heights") = axial_boundaries;
+      params.set<boundary_id_type>("bottom_boundary") = bottom_boundary;
+      params.set<boundary_id_type>("top_boundary") = top_boundary;
+
+      addMeshSubgenerator("FancyExtruderGenerator", name() + "_extruded", params);
+    }
+
+    {
+      auto params = _app.getFactory().getValidParams("RenameBoundaryGenerator");
+
+      params.set<MeshGeneratorName>("input") = name() + "_extruded";
+      params.set<std::vector<BoundaryName>>("old_boundary") = {
+          "201", "202"}; // hard coded boundary IDs in patterned mesh generator
```

Comment:
this is too low. 
I m not a fan of these hard numbers already, but at least if it s high enough we wont hit them